docs/source/routing/performance/caching/response-cache.mdx

```diff
@@ -0,0 +1,513 @@
+  enabled: true
+  debug: false # Optional debug mode for troubleshooting
+  
+  # Configure metrics collection
+  metrics:
+    enabled: true
+    ttl: 24h # How long to keep metric data
+    separate_per_type: false # Set to true for detailed per-type metrics (increases cardinality)
+  
+  # Per-subgraph configuration
+  subgraph:
+    all:
+      enabled: true
+      # PostgreSQL connection configuration
+      postgres:
+        url: "postgresql://username:password@localhost:5432/cache_db"
+        pool_size: 10 # Connection pool size (default: 10)
+        idle_timeout: "5min" # Connection idle timeout (default: 1min)
+        acquire_timeout: "50ms" # Maximum time to wait for connection (default: 50ms)
+        required_to_start: true # Fail router startup if can't connect
+        batch_size: 100 # Batch size for cache operations (default: 100)
+        cleanup_interval: "1h" # How often to clean expired entries (default: 1h)
+        
+        # Optional: TLS configuration
+        tls:
+          certificate_authorities: |
+            -----BEGIN CERTIFICATE-----
+            ...your CA certificate...
+            -----END CERTIFICATE-----
+          client_authentication:
+            certificate: |
+              -----BEGIN CERTIFICATE-----
+              ...your client certificate...
+              -----END CERTIFICATE-----
+            key: |
+              -----BEGIN PRIVATE KEY-----
+              ...your private key...
+              -----END PRIVATE KEY-----
+      
+      # Default TTL for cached responses
+      ttl: 3600s # 1 hour default
+    
+    # Per-subgraph overrides
+    subgraphs:
+      products:
+        ttl: 24h # Product data can be cached longer
+      inventory:
+        ttl: 60s # Inventory changes frequently
+      user_profiles:
+        private_id: "user_id" # Enable private caching per user
+      analytics:
+        enabled: false # Disable caching for analytics subgraph
+```
+
+### PostgreSQL connection configuration
+
+Response caching requires a PostgreSQL database. Here are common connection patterns:
+
+#### Basic connection
+
+```yaml title="router.yaml"
+response_cache:
+  subgraph:
+    all:
+      postgres:
+        url: "postgresql://myuser:mypass@localhost:5432/router_cache"
+        required_to_start: true
+```
+
+#### Connection with separate credentials
+
+```yaml title="router.yaml"
+response_cache:
+  subgraph:
+    all:
+      postgres:
+        url: "postgresql://localhost:5432/router_cache"
+        username: "cache_user"
+        password: "secure_password"
+```
+
+#### Production configuration with connection pooling
+
+```yaml title="router.yaml"
+response_cache:
+  subgraph:
+    all:
+      postgres:
+        url: "postgresql://localhost:5432/router_cache"
+        username: "cache_user"
+        password: "secure_password"
+        pool_size: 20
+        idle_timeout: "10min"
+        acquire_timeout: "100ms"
+        cleanup_interval: "2h"
+        batch_size: 200
+```
```

Comment:
We should not call it production configuration, it's very dependent on workloads, we should just call it exhaustive configuration or something like this

Reply:
"Advanced configuration"? "Optional configuration"?